converter.js

```diff
@@ -422,7 +424,7 @@ class Converter {
             }
             // Use void as return type if there were no parameters
             // Note that the join is kinda useless (see long comments above)
-            returnType = `Promise<${parameters.join(', ') || 'void'}>`
+            returnType = `Promise<${parameters.join(', ') || 'void'}>${callback.optional ? ' | void' : ''}`
```

Comment:
- something like `Promise<string, string>` isn't a promise of two values, it only takes one type parameter so it'd be invalid, i suggest using something like `object` (maybe they'll implement the api with properties named after the parameters) or just `any`
- it's considered bad practice to sometimes return a promise and sometimes something else, and i don't think any future extension api would do that (make the type in the promise optional instead)

Reply:
1. Yeah I know and that is how the generator handles those right now (the only places that happens are currently unsupported in firefox, so it's unfortunately not possible for us to test). Hence the comment about the join being useless. Or did I completely misunderstand what you meant?
2. I'm not entirely sure what you mean by making the type in the promise optional?
Something like `Promise<number?>` is a syntax error.

Reply:
1. i just meant that this is an error:
   
   ```typescript
   // [ts] Generic type 'Promise<T>' requires 1 type argument(s).
   declare const p: Promise<string, number>;
   ```
   
   the comment about `object` was that maybe they'll implement that unsupported extension api that has multiple callback parameters by using an object, possibly with properties that are named the same as the parameters to the callback (or maybe not)
   
2. i meant `Promise<number | void>`, that way it's the same as a normal optional return except wrapped in a promise

Reply:
1. Ah yes, now I understand. What do you think the best way to handle this is right now? For now it just outputs the generic type `object`.
2. I see, will fix :)